### ISSUE TITLE
sql: joinreader should ignore NULLs in input key

### ISSUE THREAD

**Issue by RaduBerinde** [cockroachdb/cockroach#27032]
The JoinReader will happily look up a NULL value which might return results. Example:

```sql
CREATE TABLE t (a INT);
INSERT INTO t VALUES (NULL);
CREATE TABLE u (x INT PRIMARY KEY, y INT, z INT, UNIQUE INDEX (y));
INSERT INTO u VALUES (1, NULL, NULL);

SELECT y FROM t, u@u_y_key WHERE a=y;
+---+
| y |
+---+
+---+

> set experimental_force_lookup_join=true;

SELECT y FROM t, u@u_y_key WHERE a=y;
+------+
|  y   |
+------+
| NULL |
+------+
```

We should shortcircuit the lookup if any of the key columns is NULL.

**Comment by RaduBerinde** [comment]
@solongordon let me know if you can work on the fix soon, otherwise I'll take a look (#27923 is blocked on this).

**Comment by solongordon** [comment]
Yes, I should be able to take a look tomorrow. Thanks for catching it.
